REGRESSION (Fullscreen API): cnn.com: Videos are zoomed in when playing in full screen on iOS

anttijk opened the pull request:
#### 9b890f5bbf67ee100a7e5962ee58168e82fa71d1
<pre>
REGRESSION (Fullscreen API): cnn.com: Videos are zoomed in when playing in full screen on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=267543">https://bugs.webkit.org/show_bug.cgi?id=267543</a>
<a href="https://rdar.apple.com/119640248">rdar://119640248</a>

Reviewed by Tim Nguyen.

The page uses bitmovin video player which correctly sets &apos;object-fit:contain&apos; for the full screen
video element. However the page also has a style rule with equivalent specificity that sets &apos;object-fit:fill&apos;.
The video player stylesheet is inserted dynamically and its position in the document varies, allowing
the &apos;fill&apos; rule win under some circumstances.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsFullscreenObjectFitQuirk const):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Adjust the style &apos;object-fit:fill&apos; -&gt; &apos;object-fit:contain&apos;

Canonical link: <a href="https://commits.webkit.org/273061@main">https://commits.webkit.org/273061@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a4bab6e3a1e9346ffc42112813cdbd4f4bbe44ef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34165 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12970 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36150 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36851 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30959 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35246 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15356 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10102 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29999 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34674 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10976 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30482 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9586 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/9690 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30516 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38139 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31040 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30830 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35784 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9810 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/7694 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33697 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11595 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7859 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10371 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10625 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -721,6 +721,21 @@ bool Quirks::needsFullscreenDisplayNoneQuirk() const
 #endif
 }
 
+bool Quirks::needsFullscreenObjectFitQuirk() const
```
Can you add this comment:
`// cnn.com rdar://119640248`

Makes @karlcow's life easier so he can more easily audit what quirks can be removed.

karlcow replied:
me or anyone in QA who's less savy with Blame and multiple rewrite history.

anttijk committed:
REGRESSION (Fullscreen API): cnn.com: Videos are zoomed in when playing in full screen on iOS
https://bugs.webkit.org/show_bug.cgi?id=267543
rdar://119640248

Reviewed by Tim Nguyen.

The page uses bitmovin video player which correctly sets 'object-fit:contain' for the full screen
video element. However the page also has a style rule with equivalent specificity that sets 'object-fit:fill'.
The video player stylesheet is inserted dynamically and its position in the document varies, allowing
the 'fill' rule win under some circumstances.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsFullscreenObjectFitQuirk const):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Adjust the style 'object-fit:fill' -> 'object-fit:contain'

Canonical link: https://commits.webkit.org/273061@main
